ext/pdo_mysql/mysql_driver.c

```diff
@@ -819,7 +819,7 @@ static int pdo_mysql_handle_factory(pdo_dbh_t *dbh, zval *driver_options)
 #ifdef PHP_WIN32
 	if (vars[2].optval && !strcmp(".", vars[2].optval)) {
 #else
-	if (vars[2].optval && !strcmp("localhost", vars[2].optval)) {
+	if (strlen(vars[2].optval) == 0 || (vars[2].optval && !strcmp("localhost", vars[2].optval))) {
```

Comment:
I think `vars[2].optval` must be checked in every cases.

I would suggest:
```suggestion
	if (vars[2].optval && (strlen(vars[2].optval) == 0 || !strcmp("localhost", vars[2].optval))) {
```